bin/i18n/resources/dashboard/course_content/sync_in.rb

```diff
@@ -155,14 +155,21 @@ def get_i18n_strings(level)
                 i18n_strings['placeholder_texts'].merge! get_all_placeholder_text_types(processed_doc)
               end
 
-              # start_html
-              if level.start_html
-                start_html = Nokogiri::XML(level.start_html, &:noblanks)
-                i18n_strings['start_html'] = Hash.new unless level.start_html.empty?
-
-                # match any element that contains text
-                start_html.xpath('//*[text()[normalize-space()]]').each do |element|
-                  i18n_strings['start_html'][element.text] = element.text
+              # start_libraries
+              unless level.start_libraries.blank?
+                level_libraries = JSON.parse(level.start_libraries)
+                level_libraries.each do |library|
+                  next unless /^i18n_/i.match?(library["name"])
+                  library_name = library["name"]
+                  library_source = library["source"]
+                  translation_text = library_source.match(/var TRANSLATIONTEXT = (\{[^}]*});/m).captures
+
+                  i18n_strings['start_libraries'] = Hash.new if i18n_strings['start_libraries'].blank?
+                  i18n_strings['start_libraries'][library_name] = {}
+
+                  JSON.parse(translation_text[0]).each do |key, value|
+                    i18n_strings['start_libraries'][library_name][key] = value
+                  end
```

Comment:
Should we check if `library['name']` and `library['source']` are present?
```suggestion
              if level.start_libraries.present?
                level_libraries = JSON.parse(level.start_libraries)
                level_libraries.each do |library|
                  library_name = library['name']
                  next if library_name.blank?
                  next unless /^i18n_/i.match?(library_name)

                  library_source = library['source']
                  next if library_source.blank?

                  translation_json = library_source[/var TRANSLATIONTEXT = (\{[^}]*});/m, 1]
                  next if translation_json.blank?

                  i18n_strings['start_libraries'] ||= {}
                  i18n_strings['start_libraries'][library_name] = {}

                  JSON.parse(translation_json).each do |key, value|
                    i18n_strings['start_libraries'][library_name][key] = value
                  end
```

Reply:
Libraries can not be exported if they have no source:
<img width="805" alt="Screenshot 2023-10-02 at 19 15 47" src="https://github.com/code-dot-org/code-dot-org/assets/66776217/c1ec88a8-2ded-40a6-89c8-8a7d365b4165">

In addition, you can not leave the library name blank:

https://github.com/code-dot-org/code-dot-org/assets/66776217/a69290ab-612b-4481-b030-f2bb5e1bd611

Reply:
Although that's true now, it's possible a legacy level might have blank values. You could do a quick query to verify this or you could add the blank checks.

Reply:
It is worth checking.

Reply:
After querying the levels with libraries and talking with @jmkulwik, Libraries are required to have a name and at least one function.
Old libraries that don't follow this format can be deleted as they would not work.